packages/is-shallow-equal/package.json

```diff
@@ -21,6 +21,7 @@
 	},
 	"files": [
 		"arrays.js",
+		"index.js",
```

Comment:
Is this module totally broken from NPM ? This seems like it would be necessary for it to work 😯 

Sometimes I wonder if it's worthwhile to use a `files` whitelist, vs. the (apparently real) risk of neglecting to include files.

**Edit:**

At least on unpkg, it seems to be intact? https://unpkg.com/@wordpress/is-shallow-equal@1.3.0/index.js

How does that work?

There are some always-included `files`, but `index.js` is not one of them:

https://docs.npmjs.com/files/package.json#files

And I don't think we'd want to cherry-pick individual "always included"; either consistently assume the inherited always-included, or consistently list them all explicitly.

Reply:
> There are some always-included `files`, but `index.js` is not one of them:
> 
> https://docs.npmjs.com/files/package.json#files

It is included, not because of the name, but because it's the file in the “main” field.

Reply:
> Certain files are always included, regardless of settings:
> 
> * package.json
> * README
> * CHANGES / CHANGELOG / HISTORY
> * LICENSE / LICENCE
> * NOTICE
> * The file in the “main” field

I can remove all files listed in `main`. I assumed it can be confusing for those who don't know how it exactly works and it's better to whitelist all code related files. I'd be also fine with using `lib` folder for those packages which use CommonJS syntax.

Reply:
> > There are some always-included `files`, but `index.js` is not one of them:
> > https://docs.npmjs.com/files/package.json#files
> 
> It is included, not because of the name, but because it's the file in the “main” field.

Ah! I searched the page for the literal term "index", and admittedly did not read the list in detail this time around 😅 That makes sense now.

I'm on the fence about whether we include it, or allow it to be implied from `main`, or include it and avoid `main`, or include _all_ implicit files. I agree it can be confusing for those unfamiliar with the default files (_me_, apparently 😆). Since generally I'm expecting these to be "the files required for runtime execution", seems reasonable to expect `index.js` here.

Reply:
Yes, I'm fine with whatever people prefer here. I can keep it as is, remove `main` from `files`. Both work for me.

Reply:
I merged this PR, we can revisit this later :)

Reply:
> I merged this PR, we can revisit this later :)

Yeah, I'm not really sure what the best option is, since they all seem to have some downside. It's fine enough to move forward with.